src/renderer/root/components/App/ButtonBar/ButtonBar.js

```diff
@@ -67,17 +68,19 @@ export default class ButtonBar extends React.Component {
     this.getBrowserWindow().close();
   }
 
-  showWindowIcons = () => {
-    return process.platform !== 'darwin';
-  }
-
-  updateIsMax = () => {
+  handleUpdateMaximized = () => {
     const win = this.getBrowserWindow();
     if (win) {
       this.setState({ isMaximized: win.isMaximized() });
```

Comment:
Mutating state here

Reply:
@DalderupMaurice It's mutating state, but it's not mutating the existing state object.  We can still use `PureComponent` because of that differentiation.  If we were updating a state object/nested object directly, then it'd be an issue.

Here's an example that might help clarify.

```js
let object = { foo: 'foo' };

// bad
object.bar = 'bar';

// good
object = { ...this.object, bar: 'bar' };
```

In the first scenario, we're mutating the object, but it's still the same object, so PureComponent would not know there is a change since it just does a shallow equality check.  In the second scenario, we're creating a new object, and so it would be seen as a new object since a shallow equality check returns false.